allennlp/semparse/worlds/text2sql_world.py

```diff
@@ -1,31 +1,58 @@
+from allennlp.semparse.contexts.text2sql_table_context import GRAMMAR_DICTIONARY
+from allennlp.semparse.contexts.text2sql_table_context import update_grammar_with_table_values
+from allennlp.semparse.contexts.text2sql_table_context import update_grammar_with_tables
 
 class Text2SqlWorld:
     """
     World representation for any of the Text2Sql datasets.
 
     Parameters
     ----------
-    sql_table_context : ``Text2SqlTableContext``
-        This defines what sort of table-based constraints we apply
-        to the world.
+    schema_path: ``str``
+        A path to a schema file which we read into a dictionary
+        representing the SQL tables in the dataset, the keys are the
+        names of the tables that map to lists of the table's column names.
+    cursor : ``Cursor``, optional (default = None)
+        An optional cursor for a database, which is used to add
+        database values to the grammar.
+    use_prelinked_entities : ``bool``, (default = True)
+        Whether or not to use the pre-linked entities from the text2sql data.
+        We take this parameter here because it effects whether we need to add
+        table values to the grammar.
     """
-    def __init__(self, sql_table_context: Text2SqlTableContext) -> None:
+    def __init__(self,
+                 schema_path: str,
+                 cursor: Cursor = None,
+                 use_prelinked_entities: bool = True) -> None:
+        self.cursor = cursor
+        self.schema = read_dataset_schema(schema_path)
+        self.use_prelinked_entities = use_prelinked_entities
+
         # NOTE: This base dictionary should not be modified.
-        self.sql_table_context = sql_table_context
-        self.base_grammar_dictionary = sql_table_context.get_grammar_dictionary()
+        self.base_grammar_dictionary = self._initialize_grammar_dictionary(deepcopy(GRAMMAR_DICTIONARY))
 
-    def get_action_sequence_and_all_actions(self, query: List[str]) -> Tuple[List[str], List[str]]:
-        # TODO(Mark): Add in modifications here
+    def get_action_sequence_and_all_actions(self,
+                                            query: List[str] = None,
+                                            prelinked_entities: Dict[str, str] = None) -> Tuple[List[str], List[str]]: # pylint: disable=line-too-long
         grammar_with_context = deepcopy(self.base_grammar_dictionary)
+
+        if not self.use_prelinked_entities and prelinked_entities is not None:
+            raise ConfigurationError("The Text2SqlWorld was specified to not use prelinked "
+                                     "entities, but prelinked entities were passed.")
+        prelinked_entities = prelinked_entities or {}
+        for token in prelinked_entities.keys():
+            grammar_with_context["value"] = [f'"\'{token}\'"'] + grammar_with_context["value"]
+
         grammar = Grammar(format_grammar_string(grammar_with_context))
```

Comment:
Generating ``Grammar`` objects was super slow for contextual ATIS because we had to generate concrete values (a production rule often had all the values in a column as the right hand side) so to make it faster we only make one call to the ``Grammar`` constructor and edit that grammar object. Probably not an issue here since anonymizing the variables makes the grammar much smaller, but we might want to change this later if we want to add an option for not doing variable anonymization.